### ISSUE TITLE
Need a basic server

### ISSUE THREAD

**Issue by stephencweiss** [stephencweiss/matilda#1]
- [x] Serve a static Index.html
- [x] Route a CRUD to DB and receive confirmation that routes are successfully being hit

**Comment by stephencweiss** [comment]
Basic GET is working. 
```
app.get('/budget-item', (request, response) => {
  db.BudgetItem.findAll({}) 
  .then(message => response.status(200).send(message))
    .catch(err => response.status(404).send(errorMessage, err));
});
```
![image](https://user-images.githubusercontent.com/39878535/49334440-051f5980-f58b-11e8-9223-6b7948757358.png)

**Comment by stephencweiss** [comment]
Specific User's GET
```
app.get('/myBudget/:budgetId', (request, response) => {
  db.BudgetItem.findAll({ where: { budget_id: request.params.budgetId } })
    .then(message => response.status(200).send(message))
    .catch(err => response.status(404).send(errorMessage, err))
})
```
![image](https://user-images.githubusercontent.com/39878535/49334485-14eb6d80-f58c-11e8-81b2-9d9215e4bdc9.png)

**Comment by stephencweiss** [comment]
POST a budgetItem
```
app.use(bodyParser.json());

app.post('/newBudgetItem', (request, response) => {
  // Assume that the budgetId is known and part of the request (body?)
  // the budgetId is defined earlier in the process because that's the page we're on.  
  db.BudgetItem.create({
    category: request.body.category,
    hours_allocated: request.body.hoursAllocated,
    budget_id: request.body.budgetId
  })
    .then(result => response.status(201).send(result))
    .catch(err => response.status(500).send(errorMessage, err));
});
```

![image](https://user-images.githubusercontent.com/39878535/49334624-4580d680-f58f-11e8-9b1c-bb5b30fcd8c7.png)

Needed to add the bodyParser to be able to use the `request.body` values

**Comment by stephencweiss** [comment]
PUT a budgetItem
```
app.put('/updateBudgetItem/:budgetItemId', (request, response) => {
  db.BudgetItem.update({
    category: request.body.category,
    hours_allocated: request.body.hoursAllocated
    },
    { where: 
      { budget_item_id: request.params.budgetItemId }
  })
    .then(message => response.status(200).send(message))
    .catch(err => response.status(404).send(errorMessage, err))
})
```
![image](https://user-images.githubusercontent.com/39878535/49334730-c6d96880-f591-11e8-8113-16943ed87b07.png)

**Comment by stephencweiss** [comment]
DELETE a budgetItem

```
app.delete('/deleteBudgetItem/:budgetItemId', (request, response) => {
  console.log(chalk.green(`Budget Id to delete -->`),request.params.budgetItemId)
  db.BudgetItem.destroy({ 
    where: { budget_item_id: request.params.budgetItemId },
    limit: 1 
  })
    .then(() => response.sendStatus(204))
    .catch(err => response.status(404).send(errorMessage, err))
})
```

![image](https://user-images.githubusercontent.com/39878535/49334793-3b60d700-f593-11e8-8d6d-8ead2b7e2458.png)

**Comment by stephencweiss** [comment]
Addressed with the PR #4
